src/MoBi.Core/Services/BuildingBlockVersionUpdater.cs

```diff
@@ -18,23 +20,38 @@ public class BuildingBlockVersionUpdater : IBuildingBlockVersionUpdater
    {
       private readonly IMoBiProjectRetriever _projectRetriever;
       private readonly IEventPublisher _eventPublisher;
+      private readonly IDialogCreator _dialogCreator;
 
-      public BuildingBlockVersionUpdater(IMoBiProjectRetriever projectRetriever, IEventPublisher eventPublisher)
+      public BuildingBlockVersionUpdater(IMoBiProjectRetriever projectRetriever, IEventPublisher eventPublisher, IDialogCreator dialogCreator)
       {
          _projectRetriever = projectRetriever;
          _eventPublisher = eventPublisher;
+         _dialogCreator = dialogCreator;
       }
 
       public void UpdateBuildingBlockVersion(IBuildingBlock buildingBlock, uint newVersion)
       {
          if (buildingBlock == null)
             return;
 
+         if (shouldShowModuleConversionWarning(buildingBlock))
+            showModuleConversionWarning(buildingBlock);
+
          buildingBlock.Version = newVersion;
          publishSimulationStatusChangedEvents(buildingBlock);
          publishModuleStatusChangedEvents(buildingBlock);
       }
 
+      private void showModuleConversionWarning(IBuildingBlock buildingBlock)
+      {
+         _dialogCreator.MessageBoxInfo(AppConstants.Captions.TheModuleWillBeConvertedFromPKSimToExtensionModule(buildingBlock.Module.Name));
+      }
+
+      private bool shouldShowModuleConversionWarning(IBuildingBlock buildingBlock)
+      {
+         return buildingBlock.Module != null && buildingBlock.Module.IsPKSimModule;
```

Comment:
Are we sure we won't get this message more than once? Seems a bit dangerous to act so deep in the call chain